### ISSUE TITLE
Rename/reword `parallel=True` option to `open_mfdataset`

### ISSUE THREAD

**Issue by gjoseph92** [pydata/xarray#6433]
### What is your issue?

Based on its name, I was surprised to find that `open_mfdataset(..., parallel=True)` computed the whole dataset eagerly, whereas `parallel=False` just returned it in dask form. (I generally think of "dask" as related to "parallel".)

I guess the [docs](https://docs.xarray.dev/en/stable/generated/xarray.open_mfdataset.html#xarray.open_mfdataset) do technically say this, but it's a bit hard to parse:
> If True, the open and preprocess steps of this function will be performed in parallel using `dask.delayed`. Default is False.

The docstring could maybe instead mention "If False (default), the data is returned in dask form. If True, it will be computed immediately (using dask), then returned in NumPy form".

More intuitive to me would be renaming the argument to `compute=False`. Or even deprecating the argument entirely and having a `load_mfdataset` function, in the same way that `load_dataset` is the eager version of `open_dataset`.

**Comment by keewis** [comment]
`parallel=True` will *open* (read the metadata) and preprocess the datasets in parallel using `dask`, which will happen in sequence with `parallel=False`. The difference is [roughly](https://github.com/pydata/xarray/blob/305533d585389f7240ae2383a323337d4761d33a/xarray/backends/api.py#L896-L916):
```python
if parallel:
    datasets, = dask.compute([dask.delayed(xr.open_dataset)(path, *args, **kwargs) for path in paths])
else:
    datasets = [xr.open_dataset(path, *args, **kwargs) for path in paths]
```
So unlike `load_dataset` / the proposed `load_mfdataset` this will read metadata but not the actual data, and the result should be the same regardless of the `parallel` parameter (if it doesn't that would be a bug).

I agree that we could improve the parameter description, though.

**Comment by dcherian** [comment]
>  open_mfdataset(..., parallel=True) computed the whole dataset eagerly

This isn't  totally true. It'll should only compute and save  when comparing coordinate variables for equality as part  of the concatenation step (yes, bad default!). See the [note here](https://docs.xarray.dev/en/stable/user-guide/io.html#reading-multi-file-datasets).

 The code is https://github.com/pydata/xarray/blob/305533d585389f7240ae2383a323337d4761d33a/xarray/core/concat.py#L340-L360
